modules/dnn/src/layers/split_layer.cpp

```diff
@@ -66,6 +72,12 @@ class SplitLayerImpl CV_FINAL : public SplitLayer
         }
     }
 
+    virtual bool supportBackend(int backendId) CV_OVERRIDE
+    {
+        return backendId == DNN_BACKEND_OPENCV ||
+              (backendId == DNN_BACKEND_CUDA && haveCUDA());
```

Comment:
@dkurt Do we really need `haveCUDA()` (or similar `haveHalide()`) checks in these calls.
If there is no CUDA, then `backendId` is not able to setup into `DNN_BACKEND_CUDA` value.

Reply:
Currently, `haveCUDA` returns if `HAVE_CUDA` is defined or not but it's possible to add more constraints to `haveCUDA` like checking if a CUDA device is present.